core/ops.rs

```diff
@@ -106,14 +107,19 @@ pub fn to_op_result<R: Serialize + 'static>(
   }
 }
 
-// TODO(@AaronO): optimize OpCtx(s) mem usage ?
+/// Per-op context.
+///
+// Note: We don't worry too much about the size of this struct because it's allocated once per realm, and is
+// stored in a contiguous array.
 pub struct OpCtx {
   pub id: OpId,
   pub state: Rc<RefCell<OpState>>,
   pub decl: Rc<OpDecl>,
   pub fast_fn_c_info: Option<NonNull<v8::fast_api::CFunctionInfo>>,
   pub runtime_state: Weak<RefCell<JsRuntimeState>>,
   pub(crate) context_state: Rc<RefCell<ContextState>>,
+  /// If the last fast op failed, stores the error to be picked up by the slow op.
+  pub(crate) last_fast_error: UnsafeCell<Option<AnyError>>,
```

Comment:
I'm not sure this is okay - what if you have an async fast op - only the dispatching path is "fast" but after dispatching the op it's handled by the event loop. It feels to me it can become racy and eg. the second dispatched op will set the error first, how would it work in such case?

Reply:
This is part of the fast API -- to dispatch an error you just need to tell it to call the fallback path and it redispatches the op with the same values, but the ability to throw an error from the scope.

You cannot end up with any code running between the part that asks for the fallback to run and the entry of the same slow call.

What I can do is add comments to both places helping to clarify this, however.

Reply:
Feel free to do it in a follow up